File: task3/Go/task3.go
// go run task3.go
package main

import (
	"fmt"
	"math"
	"os"
)

// проверка на полный квадрат
func isPerfectSquare(x int) bool {
	root := int(math.Sqrt(float64(x)))
	return root*root == x
}

func main() {
	var choice int
	fmt.Println("1 - input from console\n2 - input from file")
	fmt.Scan(&choice)

	count := 0

	if choice == 1 {
		var n, x int
		fmt.Print("Enter N: ")
		fmt.Scan(&n)

		for i := 0; i < n; i++ {
			fmt.Scan(&x)
			if isPerfectSquare(x) {
				count++
			}
		}

	} else if choice == 2 {
		var filename string
		fmt.Print("Enter file name: ")
		fmt.Scan(&filename)

		file, err := os.Open(filename)
		if err != nil {
			fmt.Println("File open error")
			return
		}
		defer file.Close()

		var n, x int
		fmt.Fscan(file, &n)

		for i := 0; i < n; i++ {
			fmt.Fscan(file, &x)
			if isPerfectSquare(x) {
				count++
			}
		}

	} else {
		fmt.Println("Invalid choice")
		return
	}

	fmt.Println("Result:", count)
}

File: task3/Java/task3.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class task3 {

    // проверка на полный квадрат
    public static boolean isPerfectSquare(int x) {
        int root = (int)Math.sqrt(x);
        return root * root == x;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("1 - input from console\n2 - input from file");
        int choice = sc.nextInt();

        int count = 0;

        if (choice == 1) {
            System.out.print("Enter N: ");
            int n = sc.nextInt();

            for (int i = 0; i < n; i++) {
                int x = sc.nextInt();
                if (isPerfectSquare(x)) {
                    count++;
                }
            }

        } else if (choice == 2) {
            System.out.print("Enter file name: ");
            String filename = sc.next();

            try {
                Scanner file = new Scanner(new File(filename));

                int n = file.nextInt();
                for (int i = 0; i < n; i++) {
                    int x = file.nextInt();
                    if (isPerfectSquare(x)) {
                        count++;
                    }
                }

                file.close();

            } catch (FileNotFoundException e) {
                System.out.println("File open error");
                return;
            }

        } else {
            System.out.println("Invalid choice");
            return;
        }

        System.out.println("Result: " + count);
    }
}